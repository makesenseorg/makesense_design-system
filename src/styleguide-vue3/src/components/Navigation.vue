<template>
  <div class="navigation">
    <themer />
    <div class="navigation-search">
      <ds-input
        v-model="searchString"
        placeholder="Filter menu ..."
        icon="search"
      />
    </div>
    <ds-menu
      @navigate="$emit('navigate')"
      :routes="routes"
      :url-parser="urlParser"
      :name-parser="nameParser"
      :is-exact="isExact"
    />
    <i18n />
  </div>
</template>

<script>
import Themer from "./Themer";
import I18n from "./I18n";
import DsInput from "./data-input/Input/Input";
import DsMenu from "./navigation/Menu/Menu";
export default {
  name: "GuideNavigation",
  components: {
    I18n,
    Themer,
    DsInput,
    DsMenu,
  },
  data() {
    return {
      searchString: "",
    };
  },
  computed: {
    routes() {
      const routes = this.$router.options.routes.filter((route) => {
        return route.path !== "*";
      });
      const newRoutes = routes
        .map((route) => {
          const [parent, ...children] = [...route.children];
          parent.path = route.path;
          parent.children = children.filter(this.fitsSearch);
          return parent;
        })
        .filter((route) => {
          return route.children.length || this.fitsSearch(route);
        });
      return newRoutes;
    },
    searchParts() {
      return this.searchString.split(" ");
    },
  },
  methods: {
    fitsSearch(route) {
      if (!this.searchString) {
        return true;
      }
      return this.searchParts.every((search) => {
        if (!search) {
          return true;
        }
        return route.name.toLowerCase().includes(search.toLowerCase());
      });
    },
    nameParser(route) {
      return this.$filters.componentName(route.name);
    },
    urlParser(route, parents) {
      return {
        path: (
          parents.map((parent) => parent.path).join("") +
          "/" +
          route.path
        ).replaceAll("//", "/"),
      };
    },
    isExact(url) {
      return url.name === "Introduction";
    },
  },
};
</script>

<style lang="scss" scoped>
.navigation {
  padding: $space-m $space-xs;
}

.navigation-search {
  padding: 0 $space-s;
  margin-bottom: $space-m;
}
</style>
