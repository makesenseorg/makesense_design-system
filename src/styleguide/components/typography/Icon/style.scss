.ds-icon {
  @include reset;
  display: inline-flex;
  align-items: center;
  vertical-align: middle;
  height: 1em;
}

.ds-icon-svg {
  line-height: 1;
  height: 1.2em;
  // Use this if the icons are build with strokes
  // stroke: currentColor;
  // stroke-width: 2.5px;
  // stroke-linejoin: miter;
  // stroke-linecap: miter;
  // overflow: visible;
  // Use this if the icons are build with solids
  fill: currentColor
}