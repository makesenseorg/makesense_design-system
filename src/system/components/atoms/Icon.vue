<template>
  <font-awesome-icon
    v-if="library === 'font-awesome'"
    :icon="getFaIcon"
    :class="`icon icon--color-${color}`"
  />
  <svg
    v-else
    :class="`feather feather-${getIcon}`"
    xmlns="http://www.w3.org/2000/svg"
    :width="size"
    :height="size"
    :viewBox="`0 0 ${defaultSize} ${defaultSize}`"
    fill="none"
    stroke="currentColor"
    :stroke-width="strokeWidth"
    stroke-linecap="round"
    stroke-linejoin="round"
    v-html="path"
  ></svg>
</template>

<script>
// todo : import dynamically (without breaking use in nuxt.js)
import feather from "feather-icons";
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

import {
  faArrowDown,
  faSort,
  faBars,
  faSearch,
  faArrowUp,
  faTimes,
  faTag,
  faArrowRight,
  faArrowLeft,
  faLock,
  faLifeRing,
  faBuilding,
  faCheck,
  faGlobeEurope,
  faClock,
  faLink,
  faWallet,
  faShare,
  faMapMarkerAlt
} from "@fortawesome/free-solid-svg-icons";

/**
 * Nice icons to make it more fun âœ¨
 * @version 0.1.0
 */
export default {
  name: "MksIcon",
  components: { FontAwesomeIcon },
  props: {
    /**
     * Type of icon depending on the library used. Prefer feather when possible
     */
    type: {
      type: String,
      default: ""
    },
    /**
     * Color of the icon amongs theme colors, inherits from parent by default.
     */
    color: {
      type: String,
      default: "inherit"
    },
    /**
     * Which library to use : "font-awesome", "feather"
     */
    library: {
      type: String,
      default: "feather"
    },
    size: {
      type: String,
      default: "24"
    },
    strokeWidth: {
      type: String,
      default: "2"
    }
  },
  data() {
    return {
      defaultSize: 24,
      path: ""
    };
  },
  async created() {
    if (this.library === "feather") {
      this.path = feather.icons[this.type];
    }
  },
  computed: {
    getIcon: function() {
      if (this.type === "close") return "x";
      if (this.type === "arrowUp") return "arrow-up";
      if (this.type === "arrowDown") return "arrow-down";
      if (this.type === "arrowLeft") return "arrow-left";
      if (this.type === "arrowRight") return "arrow-right";
      return this.type;
    },

    getFaIcon: function() {
      switch (this.type) {
        case "arrowDown":
          return faArrowDown;
        case "arrowUp":
          return faArrowUp;
        case "arrowRight":
          return faArrowRight;
        case "arrowLeft":
          return faArrowLeft;
        case "close":
          return faTimes;
        case "search":
          return faSearch;
        case "share":
          return faShare;
        case "check":
          return faCheck;
        case "sort":
          return faSort;
        case "link":
          return faLink;
        case "company":
          return faBuilding;
        case "lang":
          return faGlobeEurope;
        case "clock":
          return faClock;
        case "mapMarker":
          return faMapMarkerAlt;
        case "tag":
          return faTag;
        case "wallet":
          return faWallet;
        case "lock":
          return faLock;
        case "menu":
          return faBars;
        case "help":
          return faLifeRing;
        default:
          return faSearch;
      }
    }
  }
};
</script>

<style lang="scss" scoped>
.icon {
  color: inherit;
  vertical-align: middle;
  font-size: inherit;
  width: 1em;

  &--color-neutral {
    color: $color-neutral-70;
  }

  &--color-primary {
    color: $color-primary;
  }

  &--color-secondary {
    color: $color-secondary;
  }

  &--color-tertiary {
    color: $color-tertiary;
  }

  &--color-positive {
    color: $color-success;
  }

  &--color-warning {
    color: $color-warning;
  }

  &--color-negative {
    color: $color-danger;
  }
}
</style>
<docs>
## Feather icons (default)
List of all available icons : https://feathericons.com/

Some examples :

```jsx
<p>
  <mks-icon type="activity"></mks-icon> 
  <mks-icon type="alert-triangle"></mks-icon> 
  <mks-icon type="arrow-left"></mks-icon>
  <mks-icon type="arrow-left-circle"></mks-icon>
  <mks-icon type="bell"></mks-icon> 
  <mks-icon type="check"></mks-icon> 
  <mks-icon type="clock"></mks-icon> 
  <mks-icon type="eye-off"></mks-icon> 
  <mks-icon type="filter"></mks-icon> 
  <mks-icon type="heart"></mks-icon> 
  <mks-icon type="list"></mks-icon> 
  <mks-icon type="lock"></mks-icon>
  <mks-icon type="map-pin"></mks-icon>
  <mks-icon type="menu"></mks-icon>
  <mks-icon type="save"></mks-icon>
  <mks-icon type="star"></mks-icon> 
  <mks-icon type="thumbs-up"></mks-icon> 
  <mks-icon type="user"></mks-icon>
  <mks-icon type="zoom-in"></mks-icon>
</p>
```

## Font Awesome icons 
Font awesome icons are still available. However you have to specify to use the font-awesome library. 

Only these ones are available as of now : 

```jsx
<p>
  <mks-icon type="arrowUp" library="font-awesome"></mks-icon> 
  <mks-icon type="arrowDown" library="font-awesome"></mks-icon> 
  <mks-icon type="arrowLeft" library="font-awesome"></mks-icon>
  <mks-icon type="arrowRight" library="font-awesome"></mks-icon>
  <mks-icon type="close" library="font-awesome"></mks-icon> 
  <mks-icon type="mapMarker" library="font-awesome"></mks-icon> 
  <mks-icon type="lang" library="font-awesome"></mks-icon> 
  <mks-icon type="link" library="font-awesome"></mks-icon> 
  <mks-icon type="tag" library="font-awesome"></mks-icon> 
  <mks-icon type="company" library="font-awesome"></mks-icon> 
  <mks-icon type="search" library="font-awesome"></mks-icon> 
  <mks-icon type="check" library="font-awesome"></mks-icon>
  <mks-icon type="share" library="font-awesome"></mks-icon>
  <mks-icon type="sort"  library="font-awesome"></mks-icon>
  <mks-icon type="clock" library="font-awesome"></mks-icon>
  <mks-icon type="wallet" library="font-awesome"></mks-icon> 
  <mks-icon type="lock" library="font-awesome"></mks-icon> 
  <mks-icon type="menu" library="font-awesome"></mks-icon>
  <mks-icon type="help" library="font-awesome"></mks-icon>
</p>
```

## Colors 

Inherits from parent by default 

```jsx
<p>
  <mks-icon type="arrow-up" color="neutral"></mks-icon> 
  <mks-icon type="arrow-up" color="primary"></mks-icon> 
  <mks-icon type="arrow-up" color="secondary"></mks-icon> 
  <mks-icon type="arrow-up" color="tertiary"></mks-icon> 
  <mks-icon type="arrow-up" color="positive"></mks-icon> 
  <mks-icon type="arrow-up" color="warning"></mks-icon> 
  <mks-icon type="arrow-up" color="negative"></mks-icon> 
</p>
```
</docs>
