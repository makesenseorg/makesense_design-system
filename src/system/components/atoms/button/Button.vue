<template>
  <component
    :is="tag"
    :class="[
      `button button--type-${type} button--size-${size}`,
      { 'button--loading': loading }
    ]"
    :role="role"
    tabindex="0"
    :type="inputType"
    :disabled="disabled"
    ref="container"
    :style="{
      minWidth: $slots.loading === undefined ? minWidth + 'px' : 'auto'
    }"
    @click="onClick"
    @keyup.enter="onClick"
  >
    <mks-icon
      v-if="icon && !iconRight"
      :type="icon"
      class="button__icon button__icon--left"
      :size="size === 'round' ? 24 : 20"
    ></mks-icon>
    <!-- @slot Content of the button-->
    <slot v-if="!loading" />
    <!-- @slot Content of the button when `loading` prop is true -->
    <slot v-else name="loading">
      <mks-loading
        type="standalone"
        :inline="true"
        color="#fff"
        :size="20"
      ></mks-loading>
    </slot>
    <mks-icon
      v-if="icon && iconRight"
      :type="icon"
      class="button__icon button__icon--right"
      :size="size === 'round' ? 24 : 20"
    ></mks-icon>
  </component>
</template>
<script>
/**
 * Buttons can be used for any action or inner link.
 * @version 0.3.0
 */
export default {
  name: "MksButton",
  release: "0.3.0",
  props: {
    /**
     * Buttons can be replaced by any of these tags : "div", "a", "input", "button", "span"
     */
    tag: {
      type: String,
      default: "button",
      validator: function(value) {
        return ["div", "a", "input", "button", "span"].indexOf(value) !== -1;
      }
    },
    /**
     * If loading is set to true, the button will display the loading slot or "Processing..."
     */
    loading: {
      type: Boolean,
      default: false
    },
    /**
     * If disabled is set to true, the button will be disabled
     */
    disabled: {
      type: Boolean,
      default: false
    },
    /**
     * Can set the input type, in case of button tag (ex: "submit")
     */
    inputType: {
      type: String,
      required: false,
      default: ""
    },
    /**
     * Color of the button. "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative"
     */
    type: {
      type: String,
      default: "primary",
      validator: function(value) {
        return (
          [
            "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative",
            "text",
            "text-active"
          ].indexOf(value) !== -1
        );
      }
    },
    /**
     * Has to be a valid icon type
     */
    icon: {
      type: String
    },
    /**
     * Place the icon on the right side of the text
     */
    iconRight: {
      type: Boolean,
      default: false
    },
    /**
     * Size of the button. "default", "small", "medium" "full", "round"
     */
    size: {
      type: String,
      default: "default",
      validator: function(value) {
        return (
          ["default", "small", "medium", "full", "round", "square"].indexOf(
            value
          ) !== -1
        );
      }
    },
    /**
     * Size of the button. "default", "small", "full", "round"
     */
    role: {
      type: String,
      default: "button"
    }
  },
  data: function() {
    return {
      minWidth: 0,
      observer: null
    };
  },
  mounted() {
    this.updateMinWidth();

    this.observer = new MutationObserver(() => {
      this.updateMinWidth();
    });

    this.observer.observe(this.$refs.container, {
      attributes: true,
      childList: true,
      characterData: true,
      subtree: true
    });
  },
  beforeDestroy: function() {
    // Clean up
    this.observer.disconnect();
  },
  methods: {
    onClick() {
      /**
       * Fires after click.
       *
       * @event click
       */
      this.$emit("click");
    },
    updateMinWidth() {
      this.$nextTick(function() {
        var container = this.$refs.container;
        if (container !== undefined) this.minWidth = container.offsetWidth;
      });
    }
  }
};
</script>

<style lang="scss" scoped>
.button {
  @include cursor-pointer;
  @include border-radius;
  @include button-reset;
  @include padding-button;
  @include text-body;
  display: inline-block;
  font-weight: $font-weight-extrabold; // override text-body
  border-width: $border-width-m;
  border-style: solid;
  text-decoration: none !important;
  transition: background-color 0.2s, border-color 0.2s;
  text-align: center;

  &:focus {
    @include focus($background-color-primary);
  }

  &--size-small {
    @include border-round;
    @include padding-button-small;
  }

  &--size-medium {
    @include border-radius;
    @include padding-button-small;
  }
  &--size-full {
    width: 100%;
    text-align: center;
  }

  &--size-round {
    @include title-small;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    user-select: none;
    border-radius: $border-radius-circle;
    padding-right: 0;
    padding-left: 0;
    width: $space-xxl;
    height: $space-xxl;
    text-align: center;

    .button__icon {
      margin: 0;
    }
  }
  &--size-square {
    @include text-body;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    user-select: none;
    border-radius: $border-radius-m;
    padding-right: 0;
    padding-left: 0;
    width: $space-xl;
    height: $space-xl;
    padding: $space-xxs;
    text-align: center;
  }

  @mixin type-text {
    @include text-small;
    background-color: transparent;
    text-transform: uppercase;
    color: inherit;
    padding: 0;
    margin: 0;
    border: none;
  }

  &--type-text {
    @include type-text;
    font-weight: $font-weight-normal;
  }

  &--type-text-active {
    @include type-text;
    font-weight: $font-weight-black;
  }

  &--type-primary {
    background-color: $color-primary;
    border-color: $color-primary;
    color: $color-primary-inverse;

    &:hover {
      background-color: $color-primary-active;
      border-color: $color-primary-active;
    }
  }

  &--type-secondary {
    background-color: $color-secondary;
    border-color: $color-secondary;
    color: $color-secondary-inverse;

    &:hover {
      // todo : garder la border peut-être? ou rendre l'effet un peu plus légeer
      background-color: $color-secondary-active;
      border-color: $color-secondary-active;
    }

    &:focus {
      @include focus($color-secondary);
    }
  }

  &--type-tertiary {
    background-color: $color-tertiary;
    border-color: $color-tertiary;
    color: $color-tertiary-inverse;

    &:hover {
      // todo : garder la border peut-être? ou rendre l'effet un peu plus légeer
      background-color: $color-tertiary-active;
      border-color: $color-tertiary-active;
    }

    &:focus {
      @include focus($color-tertiary);
    }
  }

  &--type-positive {
    background-color: $color-success;
    border-color: $color-success;
    color: $color-success-inverse;

    &:hover {
      background-color: $color-success-active;
      border-color: $color-success-active;
    }

    &:focus {
      @include focus($color-success);
    }
  }

  &--type-warning {
    background-color: $color-warning;
    border-color: $color-warning;
    color: $color-warning-inverse;

    &:hover {
      background-color: $color-warning-active;
      border-color: $color-warning-active;
    }

    &:focus {
      @include focus($color-warning-active);
    }
  }

  &--type-negative {
    background-color: $color-danger;
    border-color: $color-danger;
    color: $color-danger-inverse;

    &:hover {
      background-color: $color-danger-active;
      border-color: $color-danger-active;
    }

    &:focus {
      @include focus($color-danger-active);
    }
  }

  &--type-neutral {
    color: $color-neutral-20;
    background-color: $background-color-base;
    border-color: $background-color-softest;

    &:hover {
      background-color: $background-color-softest;
    }

    &:focus {
      @include focus($background-color-softest);
    }
  }

  &:disabled,
  &[disabled] {
    opacity: 0.5;
    cursor: default;
    pointer-events: none;
  }

  &--loading {
    opacity: 0.7;
  }
}

.button__icon {
  &--left {
    margin-right: $space-xs;
  }
  &--right {
    margin-left: $space-xs;
  }
}

button {
  outline: none !important;
}
</style>
<docs>
## Colors
  ```jsx
    <mks-button tag="button">Button default</mks-button>
    <mks-button tag="button" type="secondary">Button secondary</mks-button>
    <mks-button tag="button" type="tertiary">Button tertiary</mks-button>
    <mks-button tag="button" type="neutral">Button neutral</mks-button>
    <br>
    <mks-button type="positive">Button positive</mks-button>
    <mks-button type="warning">Button warning</mks-button>
    <mks-button type="negative">Button negative</mks-button>
  ```

## Sizes
```jsx
    <mks-button size="round" icon="menu"></mks-button>
    <mks-button size="small">Button small</mks-button>
    <mks-button size="medium">Button medium</mks-button>
    <mks-button size="default">Button default</mks-button>
    <mks-button size="full">Button full</mks-button>
  ```

## With icons
```jsx
    <mks-button size="round" icon="bell"></mks-button>
    <br>
    <mks-button icon="menu" size="small">Menu</mks-button>
    <mks-button icon="menu" icon-right size="small">Menu</mks-button>
    <br>
    <mks-button icon="star">Sauvegarder</mks-button>
    <mks-button icon="star" icon-right>Sauvegarder</mks-button>
  ```

  ## States
```jsx
    <mks-button v-bind:disabled="true">Disabled</mks-button>
    <br>
    <mks-button v-bind:loading="true">
        Load 
        <template v-slot:loading>Loading...</template>
    </mks-button>
    <br>
    <mks-button v-bind:loading="true">
        Load 
    </mks-button>
    <br>
    <mks-button v-bind:loading="true" size="small">
        Load 
    </mks-button>
  ```
</docs>
