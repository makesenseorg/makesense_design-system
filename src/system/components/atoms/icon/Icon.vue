<template>
  <svg
    :class="`icon icon-${getIcon} icon--color-${color}`"
    role="img"
    xmlns="http://www.w3.org/2000/svg"
    :width="size"
    :height="size"
    :viewBox="`0 0 ${defaultSize} ${defaultSize}`"
    fill="none"
    :stroke-width="strokeWidth"
    stroke-linecap="round"
    stroke-linejoin="round"
    v-html="path"
  ></svg>
</template>

<script>
import feather from "feather-icons";
/**
 * Nice icons to make it more fun ✨ (based on feather)
 * @version 0.1.0
 */
export default {
  name: "MksIcon",
  props: {
    /**
     * Type of icon
     */
    type: {
      type: String,
      required: true
    },
    /**
     * Description of the icon for screen readers
     */
    alt: {
      type: String,
      required: false
    },
    /**
     * Color of the icon amongs theme colors, inherits from parent by default.
     */
    color: {
      type: String,
      default: "inherit",
      validator: function(value) {
        return (
          [
            "inherit",
            "primary",
            "secondary",
            "tertiary",
            "neutral",
            "positive",
            "warning",
            "negative",
            "currentColor"
          ].indexOf(value) !== -1
        );
      }
    },
    size: {
      type: [String, Number],
      default: "24"
    },
    strokeWidth: {
      type: String,
      default: "2"
    }
  },
  data() {
    return {
      defaultSize: 24
    };
  },
  computed: {
    getIcon: function() {
      if (this.type === "close") return "x";
      if (this.type === "arrowUp") return "arrow-up";
      if (this.type === "arrowDown") return "arrow-down";
      if (this.type === "arrowLeft") return "arrow-left";
      if (this.type === "arrowRight") return "arrow-right";
      return this.type;
    },
    path() {
      return `<title id="${this.type}">${this.alt || ""}</title>${
        feather.icons[this.type]
      }`;
    }
  }
};
</script>

<style lang="scss" scoped>
.icon {
  stroke: currentColor;
  vertical-align: middle;

  &--color-neutral {
    stroke: $color-neutral-70;
  }

  &--color-primary {
    stroke: $color-primary;
  }

  &--color-secondary {
    stroke: $color-secondary;
  }

  &--color-tertiary {
    stroke: $color-tertiary;
  }

  &--color-positive {
    stroke: $color-success;
  }

  &--color-warning {
    stroke: $color-warning;
  }

  &--color-negative {
    stroke: $color-danger;
  }
}
</style>
<docs>
## Icons
[List of 286 available icons](https://feathericons.com/)

Some examples :

```jsx
<p>
  <mks-icon type="activity" alt="Activités"></mks-icon> 
  <mks-icon type="alert-triangle" alt="Attention, il fait froid!"></mks-icon> 
  <mks-icon type="arrow-left" alt="A gauche"></mks-icon>
  <mks-icon type="arrow-left-circle" alt="Retourner"></mks-icon>
  <mks-icon type="bell" alt="Notifications"></mks-icon> 
  <mks-icon type="check"></mks-icon> 
  <mks-icon type="clock"></mks-icon> 
  <mks-icon type="eye-off"></mks-icon> 
  <mks-icon type="filter"></mks-icon> 
  <mks-icon type="heart"></mks-icon> 
  <mks-icon type="list"></mks-icon> 
  <mks-icon type="lock"></mks-icon>
  <mks-icon type="map-pin"></mks-icon>
  <mks-icon type="menu"></mks-icon>
  <mks-icon type="save"></mks-icon>
  <mks-icon type="star"></mks-icon> 
  <mks-icon type="thumbs-up"></mks-icon> 
  <mks-icon type="user"></mks-icon>
  <mks-icon type="zoom-in"></mks-icon>
</p>

<p>
  <mks-icon type="star" alt="petite étoile" v-bind:size="12" color="primary"></mks-icon> 
  <mks-icon type="star" alt="étoile couleur secondaire" size="50" color="secondary"></mks-icon> 
</p>
```
</docs>
