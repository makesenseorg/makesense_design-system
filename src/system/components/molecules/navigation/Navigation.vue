<template>
  <nav :class="`navigation navigation--theme-${theme}`">
    <mks-horizontal-list tag="ul" class="navigation__list">
      <li
        class="navigation__item"
        v-for="link in links"
        :key="link.label"
        @click="$emit('click', link)"
      >
        <mks-link :to="link.to" type="menu" class>{{ link.label }}</mks-link>
      </li>
    </mks-horizontal-list>
  </nav>
</template>
<script>
/**
 * Menu used in the site header.
 * @version 0.2.0
 */
export default {
  name: "MksNavigation",
  release: "0.2.0",
  props: {
    /**
     ** Array of link objects, containing a "label" and a "to". The "to" can be either a normal http URL or a vue router or nuxt router object, with path or name.
     ** {string} label, {string | object} to
     */
    links: {
      type: Array,
      required: true
    },
    theme: {
      type: String,
      default: "primary",
      validator: value => {
        return value.match(/(primary|secondary|tertiary|neutral)/);
      }
    },
    type: {
      type: String,
      default: "main",
      validator: value => {
        return value.match(/(main|secondary)/);
      }
    }
  },
  emits: ["click"]
};
/**
 * Returns the link object that was clicked
 * @event click
 */
</script>

<style lang="scss" scoped>
.navigation {
  @include text-small-black;
}

$themes: (
  "neutral": $color-neutral-70,
  "primary": $color-primary,
  "secondary": $color-secondary,
  "tertiary": $color-tertiary
);

@each $name, $variable in $themes {
  .navigation--theme-#{$name} {
    .navigation__item + .navigation__item {
      @include flex-center;
      &:before {
        @include bean($color: $variable);
      }
    }
  }
}
</style>

<docs>

This component needs an array of link objects. It automatically detects the current page and applies styling.

The link object takes a <code>label</code> and a <code>to</code> property.

<pre><code>{ label: 'Internal link', to: { name: 'namedRoute'} }
{ label: 'Internal link', to: '/path/of/route' }</code></pre>

For more information on the link object,  <mks-link to="/atoms/mkslink">see Link component</mks-link>

## Example

```jsx
<mks-navigation type="menu" theme="secondary" v-bind:links="[{label: 'Home', to: '/'}, {label: 'Citizens', to: '/molecules/mksnavigation'}, {label: 'Entrepreneurs', to: 'https://makesense.org'}, {label: 'Organizations', to: 'https://france.makesense.org'}]"></mks-navigation>
<div style="color: #484949;background:#f8f9fa;padding:1rem;">
  <mks-navigation type="menu" theme="neutral" v-bind:links="[{label: 'Jobs', to: 'http://jobs.makesense.org'}, {label: 'Agenda', to: 'https://events.makesense.org'}, {label: 'Rejoindre l\'Ã©quipe', to: 'https://makesense.org'}, {label: 'Newsletter', to: 'https://makesense.org'}]"></mks-navigation>
</div>
```
</docs>
