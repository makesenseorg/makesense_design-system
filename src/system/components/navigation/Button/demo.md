## Button types

Use a primary button to draw the users attention to important actions. Use default buttons for less important actions.

A danger button should be used only for destructive actions.

```
<ds-button>Default</ds-button>
<ds-button primary>Primary</ds-button>
<ds-button secondary>Secondary</ds-button>
<ds-button danger>Danger</ds-button>
```

## Ghost buttons

Use a ghost button for secondary actions.

```
<ds-button ghost>Default</ds-button>
<ds-button ghost primary>Primary</ds-button>
<ds-button ghost secondary>Secondary</ds-button>
<ds-button ghost danger>Danger</ds-button>
```

## Button sizes

Use different sizes to create hierarchy.

```
<ds-button size="small">Small</ds-button>
<ds-button>Base</ds-button>
<ds-button size="large">Large</ds-button>
```

## Fullwidth button

Use a fullwidth button for important actions.

```
<ds-button primary fullwidth>Fullwidth button</ds-button>
```

## Button states

A button can take different states.

```
<ds-button>Default state</ds-button>
<ds-button disabled>Disabled state</ds-button>
<ds-button hover>Hover state</ds-button>
```

## Icon buttons

Add an icon to a button to help the user identify the button's action.

```
<ds-button icon="plus" primary>Click me</ds-button>
<ds-button icon-right="plus">Click me</ds-button>
<ds-button icon="plus" primary></ds-button>
<ds-button icon="plus" ghost></ds-button>
```

## Button as links

Provide a path to the button. You can pass a url string or a Vue router path object.

```
<ds-button path="/navigation">Click me</ds-button>
<ds-button :path="{ name: 'Navigation' }">Click me</ds-button>
```