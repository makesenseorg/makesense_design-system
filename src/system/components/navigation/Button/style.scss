.ds-button {
  @include reset;
  position: relative;
  width: auto;
  overflow: visible;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  border: 0;
  cursor: pointer;
  user-select: none;
  font-size: $font-size-base;
  line-height: $line-height-base;
  font-family: $font-family-text;
  font-weight: $font-weight-bold;
  letter-spacing: $letter-spacing-large;
  display: inline-flex;
  vertical-align: middle;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  padding: 0 $space-small;
  height: $input-height;
  border-radius: $border-radius-base;
  box-shadow: $box-shadow-small-inset, $box-shadow-x-small;
  border: $input-border-size solid transparent;
  transition: color $duration-short $ease-out,
    background-color $duration-short $ease-out;

  &:before {
    position: absolute;
    content: '';
    top: $space-xxx-small;
    left: $space-xxx-small;
    right: $space-xxx-small;
    bottom: $space-xxx-small;
    border-radius: $border-radius-base;
    box-shadow: $box-shadow-inset;
    opacity: 0;
    visiblity: hidden;
    pointer-events: none;
  }

  &:after {
    position: absolute;
    content: '';
    top: $space-xxx-small;
    left: $space-xxx-small;
    right: $space-xxx-small;
    bottom: $space-xxx-small;
    border-radius: $border-radius-base;
    border: 1px dotted currentColor;
    opacity: 0;
    visiblity: hidden;
    pointer-events: none;
    transition: all $duration-short $ease-out;
  }

  &:focus {
    outline: none;
    &:after {
      opacity: 0.8;
      visibility: visible;
    }
  }

  &:active {
    &:before {
      opacity: 0.6;
    }
  }

  &::-moz-focus-inner {
    border: 0;
    padding: 0;
  }

  &:disabled {
    opacity: 0.5;
    cursor: default;
    pointer-events: none;
  }

  // Default colors
  color: $text-color-base;
  background-color: $background-color-softer;

  &:hover,
  &.ds-button-hover {
    color: $text-color-base;
    background-color: $background-color-softer-active;
  }
}

.ds-button-fullwidth {
  width: 100%;
}

.ds-button-primary {
  color: $text-color-primary-inverse;
  background-color: $background-color-primary;

  &:hover,
  &.ds-button-hover {
    color: $text-color-primary-inverse;
    background-color: $background-color-primary-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-secondary {
  color: $text-color-secondary-inverse;
  background-color: $background-color-secondary;

  &:hover,
  &.ds-button-hover {
    color: $text-color-secondary-inverse;
    background-color: $background-color-secondary-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-danger {
  color: $text-color-danger-inverse;
  background-color: $background-color-danger;

  &:hover,
  &.ds-button-hover {
    color: $text-color-danger-inverse;
    background-color: $background-color-danger-active;
  }

  &:active {
    &:before {
      opacity: 1;
    }
  }
}

.ds-button-ghost {
  color: $text-color-base;
  background-color: transparent;
  box-shadow: none;

  &:focus {
    box-shadow: none;
  }

  &:hover,
  &.ds-hover {
    color: $text-color-base;
    background-color: $background-color-soft;
  }

  &:active {
    &:before {
      opacity: 0.6;
    }
  }

  &.ds-button-primary {
    color: $text-color-primary;
  }

  &.ds-button-secondary {
    color: $text-color-secondary;
  }

  &.ds-button-danger {
    color: $text-color-danger;
  }
}

.ds-button-size-small {
  font-size: $font-size-small;
  padding: 0 $space-x-small;
  height: $input-height-small;
}

.ds-button-size-large {
  font-size: $font-size-large;
  padding: 0 $space-base;
  height: $input-height-large;
}

.ds-button-icon-only {
  width: $input-height;
  padding: 0;
  border-radius: $border-radius-rounded;

  &:before, &:after {
    border-radius: $border-radius-rounded;
  }

  &.ds-button-size-small {
    width: $input-height-small;
  }

  &.ds-button-size-large {
    width: $input-height-large;
  }
}

.ds-button-text {
  line-height: inherit;
  display: inline-block;
  white-space: nowrap;
  margin: 0 $font-space-small;

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
}