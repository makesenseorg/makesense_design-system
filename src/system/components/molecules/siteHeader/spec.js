import { shallowMount, createLocalVue } from "@vue/test-utils";
import Comp from "./SiteHeader.vue";
import Button from "../../atoms/button/Button.vue";
import Icon from "../../atoms/icon/Icon.vue";
import Navigation from "../../molecules/navigation/Navigation.vue";

const localVue = createLocalVue();
localVue.component("MksButton", Button);
localVue.component("MksIcon", Icon);
localVue.component("MksNavigation", Navigation);

describe("SiteHeader.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      localVue
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("sets the fixed class", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        fixed: true
      }
    });
    expect(wrapper.classes()).toContain("site-header--fixed");
  });

  it("disables the sidebar", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        sidebar: false
      }
    });
    const sidebarControl = wrapper.find(".site-header__sidebar-control");
    expect(sidebarControl.exists()).toBeFalsy();
  });

  it("sets the logo", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        logo: "https://makesense.org/logo.png"
      }
    });
    const logoImg = wrapper.find(".site-header__logo img");
    expect(logoImg.attributes("src")).toBe("https://makesense.org/logo.png");
  });
});
