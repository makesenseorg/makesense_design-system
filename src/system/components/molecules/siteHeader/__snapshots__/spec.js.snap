// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader.vue matches snapshot 1`] = `
<header
  class="site-header"
>
  <!---->
   
  <div
    class="site-header__top-bar"
  >
    <mks-link
      to="https://makesense.org"
    >
      Retourner au site makesense
    </mks-link>
  </div>
   
  <div
    class="site-header__bar"
  >
    <div
      class="site-header__extra site-header__extra--left"
    />
     
    <router-link
      class="site-header__main-content"
      to="/"
    >
      <div
        class="site-header__logo"
      >
        <img
          alt="Logo"
          src=""
        />
      </div>
    </router-link>
     
    <div
      class="site-header__extra site-header__extra--right"
    />
  </div>
   
  <mks-bean-menu
    class="site-header__nav"
  />
   
  <div
    class="site-header__sidebar-control"
  >
    <mks-button-stub
      inputtype=""
      size="round"
      tag="div"
      type="primary"
    >
      <mks-icon-stub
        color="inherit"
        size="24"
        strokewidth="2"
        type="menu"
      />
    </mks-button-stub>
  </div>
</header>
`;
